lib/r7_insight/host/connection.rb

```diff
@@ -188,9 +188,11 @@ def open_connection
             ssl_context = OpenSSL::SSL::SSLContext.new
             ssl_context.cert_store = cert_store
 
-            ssl_context.min_version = OpenSSL::SSL::TLS1_1_VERSION
-            # For older versions of openssl (prior to 1.1.1) missing support for TLSv1.3
-            ssl_context.max_version = defined?(OpenSSL::SSL::TLS1_3_VERSION) ? OpenSSL::SSL::TLS1_3_VERSION : OpenSSL::SSL::TLS1_2_VERSION
+            if OpenSSL::VERSION >= "2.1.0"
+              ssl_context.min_version = OpenSSL::SSL::TLS1_1_VERSION
+              # For older versions of openssl (prior to 1.1.1) missing support for TLSv1.3
+              ssl_context.max_version = defined?(OpenSSL::SSL::TLS1_3_VERSION) ? OpenSSL::SSL::TLS1_3_VERSION : OpenSSL::SSL::TLS1_2_VERSION
+            end
```

Comment:
This is quite a shallow check. Probably quite error prone.
May I recommend this? More explicit
```suggestion
            if defined?(OpenSSL::SSL::SSLContext.method_defined? :min_version)
              ssl_context.min_version = OpenSSL::SSL::TLS1_1_VERSION
            end
            if defined?(OpenSSL::SSL::SSLContext.method_defined? :max_version)
              # For older versions of openssl (prior to 1.1.1) missing support for TLSv1.3
              ssl_context.max_version = if defined?(OpenSSL::SSL::TLS1_3_VERSION)
                                          OpenSSL::SSL::TLS1_3_VERSION
                                        else
                                          OpenSSL::SSL::TLS1_2_VERSION
                                        end
            end
```